### PR TITLE
Towards #1159: Expand CI to more OSes & Py versions

### PR DESCRIPTION
**Pull request by alexcos20**
Towards #1159

Changes proposed in this PR:

- add weekly dependabot
- add build tests on different os: windows/macos/linux
- add build tests on different python versions: 3.8 , 3.10, 3.11

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        version: ['3.8', '3.10', '3.11']
+    steps:
+      - name: Setup Ocean.py
+        uses: actions/checkout@v2
+      - name: Set up Python 3.8
```
Comment (kremalicious) [inline .github/workflows/build.yml:28]: this should work:

```suggestion
      - name: Set up Python ${{ matrix.version }}
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        version: ['3.8', '3.10', '3.11']
+    steps:
+      - name: Setup Ocean.py
+        uses: actions/checkout@v2
```
Comment (kremalicious) [inline .github/workflows/build.yml:27]: current version is v3 https://github.com/actions/checkout

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
```
Comment (kremalicious) [inline .github/workflows/build.yml:29]: current version is v4 https://github.com/actions/setup-python

Comment (alexcos20) [reply .github/workflows/build.yml]: updated, thanks!

File: .github/workflows/build.yml (hunk 2 above)
Comment (alexcos20) [reply .github/workflows/build.yml]: updated, thanks!

**Comment by trentmc** [comment]
What's the status of this? It's 6 days old. It would be very useful to have this.

**Comment by trentmc** [comment]
Not all new CI tests passed: it failed on macos-latest x python 3.10 with a "coincurve" error.

**Comment by AnaLoznianu** [comment]
Yes, @calina-c is working on fixing that failing test 😄 We got this

**Pull request merged**